packages/cspell-lib/src/Settings/CSpellSettingsServer.ts

```diff
@@ -246,9 +247,19 @@ const testNodeModules = /^node_modules\//;
 
 function resolveFilename(filename: string, relativeTo: string) {
     if (testNodeModules.test(filename)) {
-        filename = require.resolve(filename.replace(testNodeModules, ''));
+        filename = filename.replace(testNodeModules, '');
+    }
+
+    try {
+        return resolveFrom(relativeTo, filename);
+    } catch (error) {
+        if (filename.startsWith('.')) {
+            // Failed to resolve a relative module request
+            throw error;
+        }
+        // Try to resolve as a relative module
+        return resolveFrom(relativeTo, `./${filename}`);
```

Comment:
It is better to use `path.join` for windows machines. `path.join('.', filename)`

Reply:
If you run `path.join('.', filename)` it does not become a relative path. It just returns the value of the second argument. So `path.join('.', 'import.json')` just returns `'import.json'`. Not `'./import.json'` as expected.

I changed to the following instead to normalize the relative path:

```ts
  const normalizedFilename = path.sep === '/' ? filename  : filename.split(path.sep).join('/')
  return resolveFrom(relativeTo, `./${normalizedFilename}`);
```